metals/src/main/scala/scala/meta/internal/metals/DocumentSymbolProvider.scala

```diff
@@ -93,7 +93,17 @@ class DocumentSymbolProvider(trees: Trees) {
         case t: Pkg =>
           addChild(t.ref.syntax, SymbolKind.Package, t.pos, t.ref.pos, "")
           newOwner()
-        case _: Source | _: Template =>
+        case t: Term.NewAnonymous =>
+          val (name, selection) = t.templ.inits match {
+            case Nil => ("(anonymous)", t.pos)
+            case inits =>
+              (inits.map(_.tpe.syntax).mkString(" with "), inits.head.pos)
```

Comment:
I'm definitely not sure about that `inits.heads.pos`

Reply:
You can do `inits.headOption.fold(t.pos)(_.pos)` to sleep good at night. Feel free to merge!

Reply:
yeah, the doubt is mostly about the position rather than the code itself.

I'll merge anyway, we can always fix it later.